runtime/tests/checker/account_test.go

```diff
@@ -1383,3 +1383,38 @@ func TestAuthAccountContracts(t *testing.T) {
 		assert.IsType(t, &sema.AssignmentToConstantMemberError{}, errors[1])
 	})
 }
+
+func TestPublicAccountContracts(t *testing.T) {
```

Comment:
Could you please add tests that attempt to call mutating functions, i.e `add`, `update`, and `remove`, and fail? Just to make sure that we don't accidentally allow that (now or in the future)

Reply:
Thats a good idea! Added in 3cd8f8f